addons/account/wizard/account_payment_register.py

```diff
@@ -607,6 +613,18 @@ def _convert_to_wizard_currency(self, installments):
                 total_amount += comp_curr._convert(amount_residual, wizard_curr, self.company_id, self.payment_date)
         return total_amount
 
+    @api.depends('batches')
+    def _compute_can_edit_payment_method_line(self):
+        for wizard in self:
+            wizard.can_edit_payment_method_line = True
+            if len(wizard.batches) > 1:
+                previous_batch_payment_type = wizard.batches[0]['payment_values']['payment_type']
+                for batch in wizard.batches[1:]:
+                    if batch['payment_values']['payment_type'] != previous_batch_payment_type:
+                        wizard.can_edit_payment_method_line = False
+                        break
+                    previous_batch_payment_type = batch['payment_values']['payment_type']
```

Comment:
Isn't something like this more readable?
```suggestion
            wizard.can_edit_payment_method_line = len(wizard.batches) <= 1 or all(
                batch['payment_values']['payment_type'] == wizard.batches[0]['payment_values']
                for batch in wizard.batches[1:]
            )
```